### PR TITLE
Document the `max_diff` arg more in the docs to explain higher-order derivatives

### PR DESCRIPTION
**Pull request by antalszava**
Explains the usage of the `max_diff` arg more in the docs to explain higher-order derivatives with an example.

### CODE REVIEW

**Review by josh146 — APPROVED** [review]
Looks good to me @antalszava! only minor suggestion, could be worth clarifying that `max_diff` _only_ applies to gradient transform, and not backprop or adjoint

**Review by albi3ro — COMMENTED** [review]
Might be worth adding something like this to the "Gradients and Training" quickstart as well.  Much more likely to be read there.

**Comment by antalszava** [comment]
Thanks @albi3ro! I've copied the section to have it on that page too.
